src/main/java/com/salesforce/cdp/queryservice/util/TokenHelper.java

```diff
@@ -77,13 +75,19 @@ private TokenHelper() {
     }
 
     private static Map<String, String> retrieveTokenWithPasswordGrant(Properties properties, OkHttpClient client) throws SQLException {
+        // Convert password to byte array as per SA
+        if (properties.getProperty(Constants.PD) == null) {
+            throw new SQLException("Password cannot be null");
+        }
+        byte [] passwordBytes = properties.getProperty(Constants.PD).getBytes();
```

Comment:
will it work if it is of type String or byte[] ?

Reply:
So the flow is this. The client always sends password as String in properties. We convert that to byte array and once the token is exchanged we modify properties file to byte 0 in finally block.